Develop1

BadGreyFox opened the pull request:
Добавлены тесты для всех неисполняемых классов, кроме Database

BadGreyFox committed:
Добавлены Юнит-тесты для всех неисполняемых классов.

BadGreyFox committed:
Добавлены Юнит-тесты для всех неисполняемых классов.

BadGreyFox committed:
Добавлены Юнит-тесты для всех неисполняемых классов.

pomegranate282922 commented on src/test/java/praktikum/BunTest.java:
```diff
@@ -0,0 +1,25 @@
+    }
+    @Test
+    public void testGetPrice() {
+        assertEquals(expectedPrice, bun.getPrice(), 0.001);
```
Дельту лучше вынести как константу и переиспользовать 
Можно улучшить

pomegranate282922 commented on src/test/java/praktikum/BunTest.java:
```diff
@@ -0,0 +1,25 @@
+    public void testGetPrice() {
+        assertEquals(expectedPrice, bun.getPrice(), 0.001);
+    }
+}
```
Здесь ждем параметризацию 
Нужно исправить

pomegranate282922 commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,77 @@
+        when(bun.getPrice()).thenReturn(3.0f);
+        when(sauce.getPrice()).thenReturn(0.3f);
+        when(filling.getPrice()).thenReturn(0.7f);
+    }
```
Отлично что уместно применены моки

pomegranate282922 commented on src/test/java/praktikum/IngredientTest.java:
```diff
@@ -0,0 +1,27 @@
+import static org.junit.Assert.assertEquals;
+public class IngredientTest {
+
+    private static final String expectedName = RandomStringUtils.randomAlphabetic(9);
```
Статичные константы именуем заглавными буквами 
Нужно исправить

pomegranate282922 commented on src/test/java/praktikum/IngredientTest.java:
```diff
@@ -0,0 +1,27 @@
+        assertEquals("Цена не соответствует ожидаемому", expectedPrice, ingredient.getPrice(), 0.001);
+    }
+
+}
```
Нет тестов на enum 
Нужно исправить

pomegranate282922 reviewed (commented):
Работа хорошая! Применены моки и параметризация 
Замечания 
1. Использовать параметризацию в тестах на булочку 
2. Добавить тест на перечисление 
3. Нейминг

BadGreyFox committed:
test-fix: исправлены переменные для IngredientTest, добавлены константы в тесты для случайных данных из рандомайзера.

test-new: добавлены тесты для enum IngredientType, Database, и параметризованный тест для Bun.

feat: добавлены классы для константы-дельты и рандомайзер для имен и стоимости.

pomegranate282922 commented on src/test/java/praktikum/BunParameterizedTest.java:
```diff
@@ -0,0 +1,49 @@
+        return new Object[][]{
+                {"black bun", 100f},
+                {"white bun", 200f},
+                {"red bun", 300f},
```
Лучше данные взять из разных классов эквивалентности 
Можно улучшить

pomegranate282922 reviewed (approved):
Отлично, зачет!
